examples/scripts/learning_regularization_parameter_maps_for_tv_reconstruction.py

```diff
@@ -0,0 +1,876 @@
+    where :math:`A` is the forward linear operator, :math:`\nabla` is the gradient operator,
+    and :math:`\Lambda_{\theta}` is a strictly positive regularization parameter map that is estimated from
+    an input image with a network :math:`u_{\theta}` with trainable parameters :math:`\theta`.
+
+    N.B. The entire network sticks to the convention of MRpro, i.e. we work with images and k-space data
+    of shape (other*, coils, z, y, x). However, because here showcase the method for 2D problems,
+    some processing steps are necessary within the forward method. In particular, we restrict this example,
+    to be used with z=1.
+    """
+
+    def __init__(
+        self,
+        img_shape: mrpro.data.SpatialDimension[int],
+        k_shape: mrpro.data.SpatialDimension[int],
+        lambda_map_network: torch.nn.Module,
+        n_iterations: int = 128,
+    ):
+        r"""Initialize Adaptive TV Network.
+
+        Parameters
+        ----------
+        img_shape
+            image shape, i.e. the shape of the domain of the fourier operator.
+        k_shape
+           k-space shape, i.e. the shape of the domain of the fourier operator
+        lambda_map_network
+            a network that predicts a regularization parameter map from the input image.
+        n_iterations
+            number of iterations for the unrolled primal dual hybrid gradient (PDHG) algorithm.
+
+        """
+        super().__init__()
+
+        self.img_shape = img_shape
+        self.k_shape = k_shape
+
+        self.lambda_map_network = lambda_map_network
+        self.n_iterations = n_iterations
+
+        finite_differences_operator = mrpro.operators.FiniteDifferenceOp(dim=(-2, -1), mode='forward')
+        gradient_operator = (
+            mrpro.operators.RearrangeOp('grad batch ... -> batch grad ... ') @ finite_differences_operator
+        )
+        self.gradient_operator = gradient_operator
+
+        self.g = mrpro.operators.functionals.ZeroFunctional()
+
+        # operator norm of the stacked operator K=[A, \nabla]^T
+        self.stacked_operator_norm = 3.0  # analytically calculated
+
+    def estimate_lambda_map(self, image: torch.Tensor) -> torch.Tensor:
+        """Estimate regularization parameter map from image."""
+        # squeeze the dimensions that are one. In particular, the initial image
+        # is a coil-combined image (i.e. coils=1) and because, we only consider 2D problems here,
+        # z can be assumed to be always 1.
+        # (other*, coils, z, y, x) -> (other*, y, x)
+        input_image = image.abs().squeeze(-3)
+        regularization_parameter_map = self.lambda_map_network(input_image).unsqueeze(-3).unsqueeze(-3)
+        return regularization_parameter_map
+
+    def forward(
+        self,
+        initial_image: torch.Tensor,
+        csm: torch.Tensor,
+        kdata: torch.Tensor,
+        traj: mrpro.data.KTrajectory,
+        regularization_parameter: torch.Tensor | None = None,
```

Comment:
Remove. Instead, allow to set a constant parameter in init instead of a prediction network.
As this bypasses the whole network just to get a comparison method, it is a bit strange. 
Or consider a separate function for the comparison.

Reply:
please remove.